### PR TITLE
Fix building LOD's crash

### PR DESCRIPTION
**Pull request by TheNormalnij**
Fixes crash when building LOD's cause crashes after calling `engineSetPoolCapacity` or `restoreAllGameBuildings` or `RemoveAllGameBuildings`

### CODE REVIEW

File: Client/mods/deathmatch/logic/CClientBuildingManager.cpp [hunk 1]
```diff
@@ -98,9 +98,22 @@ void CClientBuildingManager::DestroyAllForABit()
 
 void CClientBuildingManager::RestoreDestroyed()
 {
-    for (CClientBuilding* building : GetBuildings())
+    bool hasInvalidLods = true;
+    while (hasInvalidLods)
```
Comment (tederis) [inline Client/mods/deathmatch/logic/CClientBuildingManager.cpp:102]: How many iterations does it usually takes? Isn't it possible that this loop become infinite in some conditions(due to model loading failure for example)?

Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/CClientBuildingManager.cpp:102]: I expect 2 iterations:
1. Generic buildings
2. Lod's

yeah, there is one situation, where `CClientBuilding::Create` doesn't work.

Comment (tederis) [reply Client/mods/deathmatch/logic/CClientBuildingManager.cpp:102]: Then maybe limit it by adding a maximum number of iterations?

**Comment by TheNormalnij** [comment]
No, i'll delete invalid elements

пт, 28 июн. 2024 г., 06:44 TEDERIs ***@***.***>:

File: Client/mods/deathmatch/logic/CClientBuildingManager.cpp (hunk 1 above)
Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/CClientBuildingManager.cpp:102]: No, i'll delete invalid elements

Comment (Dutchman101) [reply Client/mods/deathmatch/logic/CClientBuildingManager.cpp:102]: > No, i'll delete invalid elements

@TheNormalnij Do you mean you'll add a follow up commit for that?

Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/CClientBuildingManager.cpp:102]: yes

Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/CClientBuildingManager.cpp:102]: fixed

Comment (Dutchman101) [reply Client/mods/deathmatch/logic/CClientBuildingManager.cpp:102]: > fixed

@TheNormalnij
I don't see a commit when you said "fixed" ? Forgot to push it to your PR?

Comment (TheNormalnij) [reply Client/mods/deathmatch/logic/CClientBuildingManager.cpp:102]: Yes, you are right. Thank you!

**Pull request merged**